GDJS/Runtime/pixi-renderers/layer-pixi-renderer.js

```diff
@@ -252,59 +257,101 @@ gdjs.LayerPixiRenderer.prototype.enableEffect = function(name, value) {
  * @param {string} name The effect name
  * @return {boolean} true if the filter is enabled
  */
-gdjs.LayerPixiRenderer.prototype.isEffectEnabled = function(name) {
+gdjs.LayerPixiRenderer.prototype.isEffectEnabled = function (name) {
   var filter = this._filters[name];
   if (!filter) return false;
 
   return gdjs.PixiFiltersTools.isEffectEnabled(filter);
 };
 
-gdjs.LayerPixiRenderer.prototype.updateRenderTexture = function() {
-  if(!this._renderTexture) {
+gdjs.LayerPixiRenderer.prototype.updateRenderTexture = function () {
+  if (!this._renderTexture) {
     var width = this._pixiRenderer.screen.width;
     var height = this._pixiRenderer.screen.height;
     var resolution = this._pixiRenderer.resolution;
     this._renderTexture = PIXI.RenderTexture.create({
       width,
       height,
-      resolution
+      resolution,
     });
     this._renderTexture.baseTexture.scaleMode = PIXI.SCALE_MODES.LINEAR;
   }
 
-  if(this._oldWidth !== this._pixiRenderer.screen.width || 
-    this._oldHeight !== this._pixiRenderer.screen.height) {
-      this._renderTexture.resize(
-        this._pixiRenderer.width,
-        this._pixiRenderer.height
-      )
-      this._oldWidth = this._pixiRenderer.width;
-      this._oldHeight = this._pixiRenderer.height;
+  if (
+    this._oldWidth !== this._pixiRenderer.screen.width ||
+    this._oldHeight !== this._pixiRenderer.screen.height
+  ) {
+    this._renderTexture.resize(
+      this._pixiRenderer.screen.width,
+      this._pixiRenderer.screen.height
+    );
+    if (this._pixiContainer.filters) {
+      this._pixiContainer.filterArea = this._pixiRenderer.screen;
+    }
+    this._oldWidth = this._pixiRenderer.screen.width;
+    this._oldHeight = this._pixiRenderer.screen.height;
   }
 
   var oldRenderTexture = this._pixiRenderer.renderTexture.current;
   var oldSourceFrame = this._pixiRenderer.renderTexture.sourceFrame;
 
   this._pixiRenderer.renderTexture.bind(this._renderTexture);
-  this._pixiRenderer.renderTexture.clear([0.25, 0.25, 0.25, 1.0]);
+  //TODO: Pass data for clear color.
+  this._pixiRenderer.renderTexture.clear([0.5, 0.5, 0.5, 1.0]);
 
   this._pixiRenderer.render(this._pixiContainer, this._renderTexture, false);
-  this._pixiRenderer.renderTexture.bind(oldRenderTexture, oldSourceFrame, undefined);
-}
+  this._pixiRenderer.renderTexture.bind(
```

Comment:
Can you explain why this bind method call is needed?

Reply:
Since webgl/opengl is a state machine, there's only one binding point for framebuffers. (Pixi's renderTexture is just a framebuffer with a color attachment). Now, it could very well be the case that something else was bounded before we bind the renderTexture and we should, therefore, leave it to the correct state (ie, bind the `oldRenderTexture` again, it might be `null` in most of the cases, but we don't care, we've left the binding point to the state it initially had).